### ISSUE TITLE
Import (e.g. ods) doesn't respect database default collation

### ISSUE THREAD

**Issue by mokraemer** [phpmyadmin/phpmyadmin#16936]
having a database with CHARSET=utf8 COLLATE utf8_unicode_ci and importing some data from a file e.g.an ods file imports all tables as utf8_general_ci.
Since mysql is very picky on combining tables with different collations, all tables including all columns have to be converted by "hand".
Since we have a default on database base, this default should be used as well.

**Comment by williamdes** [comment]
Hi @mokraemer 

> having a database with CHARSET=utf8 COLLATE utf8_unicode_ci and importing some data from a file e.g.an ods file imports all tables as utf8_general_ci.

To conclude the import should use the collation of the database and not let the default one

> Since mysql is very picky on combining tables with different collations, all tables including all columns have to be converted by "hand".

You have an option to convert all columns in one action on the settings page of the table ;)

What version are you using ?

**Comment by mokraemer** [comment]
> You have an option to convert all columns in one action on the settings page of the table ;)
I know - but if you are importing 30 tables, this is still some work.
And if you forget to mark "convert all columns" you have to change the charset to sth. else and back before phpmyadmin accepts this option.

>What version are you using ?
5.0.4 - I tried 5.1, but this version was to buggy, so I had to revert back to 5.0.4

**Comment by williamdes** [comment]
> 5.0.4 - I tried 5.1, but this version was to buggy, so I had to revert back to 5.0.4

Could you please try the latest 5.1 version in development [(phpMyAdmin 5.1+snapshot)](https://www.phpmyadmin.net/downloads/#snapshot_5.1+snapshot) and let me know ?

This is very important for us that all bugs are fixed, if nobody reports them how can they be fixed ;)

**Comment by mokraemer** [comment]
Yeah, I know. I'm a developper too :) but sometimes you don't have enough time to fix and report all stuff and sometimes sth. has to "just work".
I'll do some more testing next week - but I think this one is still valid even in the lastest release :)

**Comment by mokraemer** [comment]
sorry for the late reply.
In 5.1.1 this is still valid. The table is created with the correct collation, but every column gets an alternate collation.
=> the database default is utf8_unicode_ci each created table has a default to utf8_unicode_ci but the varchar cols get utf8_general_ci

**Comment by williamdes** [comment]
> sorry for the late reply.
> In 5.1.1 this is still valid. The table is created with the correct collation, but every column gets an alternate collation.
> => the database default is utf8_unicode_ci each created table has a default to utf8_unicode_ci but the varchar cols get utf8_general_ci

Thank you for the feedback, I will assign it to next 5.1

**Comment by mokraemer** [comment]
can I do sth about this?
I think the main problem is, that the import adds "collation" / "charset" to the table create statement. If this was omitted, the table would be created as desired by the default of the database.

**Comment by williamdes** [comment]
> can I do sth about this? I think the main problem is, that the import adds "collation" / "charset" to the table create statement. If this was omitted, the table would be created as desired by the default of the database.

We could maybe add an option to not export collations so it imports using the default of the database or server

**Comment by mokraemer** [comment]
This report is about ods/csv import - so there is no export of collations.
The collations are added automatically on import and did not use the databases default settings.

**Comment by williamdes** [comment]
> This report is about ods/csv import - so there is no export of collations. The collations are added automatically on import and did not use the databases default settings.

That's true

**Comment by williamdes** [comment]
As I was working on this one, I found out about https://github.com/phpmyadmin/phpmyadmin/pull/19035
And I was wondering if we should consider this issue to be resolved in 6.0 and let it how it is for 5.2
@kamil-tekiela what do you think ?

**Comment by mokraemer** [comment]
Using Version 5.2.1. This bug is still valid.

**Comment by williamdes** [comment]
> Using Version 5.2.1. This bug is still valid.

Thank you for the feedback, it will be fixed in 5.2.2
